deployment/client.dc.json

```diff
@@ -0,0 +1,133 @@
+{
+  "apiVersion": "v1",
+  "kind": "Template",
+  "labels": {
+    "app": "${APP_NAME}",
+    "template": "${APP_NAME}-client"
+  },
+  "parameters": [
+    {
+      "name": "NAMESPACE",
+      "description": "The OpenShift namespace in which to deploy the client",
+      "required": true
+    },
+    {
+      "name": "APP_NAME",
+      "description": "App name that OpenShift objects will be prefixed with",
+      "required": true
+    }
+  ],
+  "objects": [
+    {
+      "apiVersion": "v1",
+      "kind": "Service",
+      "metadata": {
+        "name": "${APP_NAME}-client"
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "web",
+            "port": 3000,
+            "targetPort": 3000
+          }
+        ],
+        "selector": {
+          "name": "${APP_NAME}-client"
+        }
+      }
+    },
+    {
+      "apiVersion": "v1",
+      "kind": "Route",
+      "metadata": {
+        "name": "${APP_NAME}-client"
+      },
+      "spec": {
+        "host": "",
+        "to": {
+          "kind": "Service",
+          "name": "${APP_NAME}-client"
+        }
+      }
+    },
+    {
+      "apiVersion": "v1",
+      "kind": "DeploymentConfig",
+      "metadata": {
+        "name": "${APP_NAME}-client"
+      },
+      "spec": {
+        "replicas": 1,
+        "selector": {
+          "name": "${APP_NAME}-client"
+        },
+        "strategy": {
+          "type": "Rolling"
+        },
+        "template": {
+          "metadata": {
+            "labels": {
+              "name": "${APP_NAME}-client"
+            },
+            "name": "${APP_NAME}-client"
+          },
+          "spec": {
+            "containers": [
+              {
+                "env": [],
+                "image": " ",
+                "livenessProbe": {
+                  "httpGet": {
+                    "path": "/",
+                    "port": 3000
+                  },
+                  "initialDelaySeconds": 30,
+                  "timeoutSeconds": 3
+                },
+                "name": "${APP_NAME}-client",
+                "ports": [
+                  {
+                    "containerPort": 3000
+                  }
+                ],
+                "readinessProbe": {
+                  "httpGet": {
+                    "path": "/",
+                    "port": 3000
+                  },
+                  "initialDelaySeconds": 3,
+                  "timeoutSeconds": 3
+                },
+                "resources": {
+                  "limits": {
```

Comment:
again we should define the minimum required requests for cpu and memory.  For  a static site server  it shouldn't be t hat much i wouldn't think but measurements  are needed